bundles/io/org.openhab.io.certificate.generator/src/main/java/org/openhab/io/certificate/generator/internal/CertificateGenerator.java

```diff
@@ -0,0 +1,198 @@
+        String keystorePath = System.getProperty(JETTY_KEYSTORE_PATH_PROPERTY);
+        keystoreFile = new File(keystorePath);
+        KeyStore keyStore = KeyStore.getInstance(KEYSTORE_JKS_TYPE);
+        if (!keystoreFile.exists()) {
+            try {
+                logger.debug("No keystore found. Creation of {}", keystoreFile.getAbsolutePath());
+                boolean newFileCreated = keystoreFile.createNewFile();
+                if (newFileCreated) {
+                    keyStore.load(null, null);
+                } else {
+                    throw new IOException("Keystore file creation failed.");
+                }
+            } catch (IOException | NoSuchAlgorithmException | CertificateException e) {
+                throw new KeyStoreException("Failed to create the keystore " + keystoreFile.getAbsolutePath(), e);
+            }
+        } else {
+            try (InputStream keystoreStream = new FileInputStream(keystoreFile);) {
+                logger.debug("Keystore found. Trying to load {}", keystoreFile.getAbsolutePath());
+                keyStore.load(keystoreStream, KEYSTORE_PASSWORD.toCharArray());
+            } catch (NoSuchAlgorithmException | CertificateException | IOException e) {
+                throw new KeyStoreException("Failed to load the keystore " + keystoreFile.getAbsolutePath(), e);
+            }
+        }
+
+        return keyStore;
+    }
+
+    /**
+     * Check if the keystore contains a certificate with the KEYSTORE_ENTRY_ALIAS.
+     *
+     * @param keystore
+     * @return true if the alias is already present in the keystore, else false.
+     * @throws KeyStoreException If the keystore cannot be read.
+     */
+    private boolean isCertificateInKeystore(KeyStore keystore) throws KeyStoreException {
+        return keystore.getCertificate(KEYSTORE_ENTRY_ALIAS) != null;
+    }
+
+    /**
+     * Generate a new certificate and store it in the given keystore.
+     *
+     * @param keystore
+     * @throws CertificateException if the certificate generation has failed.
+     * @throws KeyStoreException If save of the keystore has failed.
+     */
+    private void generateCertificate(KeyStore keystore) throws CertificateException, KeyStoreException {
+        try {
+            long startTime = System.currentTimeMillis();
+            org.bouncycastle.jce.spec.ECParameterSpec ecSpec = ECNamedCurveTable.getParameterSpec(CURVE_NAME);
+            ECField field = new ECFieldFp(ecSpec.getCurve().getField().getCharacteristic());
+            EllipticCurve curve = new EllipticCurve(field, ecSpec.getCurve().getA().toBigInteger(),
+                    ecSpec.getCurve().getB().toBigInteger());
+            ECPoint pointG = new ECPoint(ecSpec.getG().getXCoord().toBigInteger(),
+                    ecSpec.getG().getYCoord().toBigInteger());
+            ECParameterSpec spec = new ECParameterSpec(curve, pointG, ecSpec.getN(), ecSpec.getH().intValueExact());
```

Comment:
This method does not exist in Java 1.7, see https://docs.oracle.com/javase/8/docs/api/java/math/BigInteger.html#intValueExact